ext/ffi/lib.rs

```diff
@@ -824,6 +873,32 @@ where
           return Err(type_error("Invalid FFI pointer type, expected null, BigInt, ArrayBuffer, or ArrayBufferView"));
         }
       }
+      NativeType::Struct(_) => {
+        if let Ok(value) = v8::Local::<v8::ArrayBufferView>::try_from(value) {
+          let byte_offset = value.byte_offset();
+          let backing_store = value
```

Comment:
Definitely an optional thing, but it might be considered if the buffer should be detached as the struct gets "passed" to the other side of the FFI API.

Of course, there's nothing forcing us to do that, and it would make certain optimisations impossible (like using a single buffer as an out pointer and parameter both, creating from and passing by value structs to FFI without any copies). It would just be a forceful way to tell the user that structs usually move their ownership when used as parameters.

Reply:
Once a buffer is used for struct by value its basically recipe for UB since memory is freed after call so I guess it'll make sense to just detach. But I'm not sure how detaching will work in ops (for async functions).

Reply:
The memory is not getting freed, I don't think. Pass-by-value is just copying after all, it doesn't steal ownership of the underlying memory. Rust "moves" a struct out of scope, disallowing touching it after pass-by-value but for instance C does not.
```rs
fn move_struct() {
  let my_struct = (0, 1, 2, vec![1, 2, 3]);
  call_with_struct(my_struct);
  my_struct.0; // Cannot access value, value is moved out of scope
}
```

```C
struct card {
  int value ;
}

typedef struct card Card;

void passByValue(Card card) {
  card.value = 5;
}

int main(void) {
  Card c ;
  c.value = 3;
  passbyValue(c);
  printf("c.value after passing: %d\n", c.value); // Compiles, prints 3 as `Card c` was copied
}
}
```

Here in the FFI the Rust code is not taking ownership of the backing_store and will not free the memory. So, in effect the FFI is more in line with C. ie. There's no direct need to detach. It's only if we want to align more with Rust that the detaching is useful.

For async functions the backing_store would need to be kept alive by the Rust code, essentially meaning it gets returned from the args parsing somehow (vec?) and then only drops when the async thread handle rejoins.